tools/build_cache.py

```diff
@@ -38,5 +39,14 @@
         print(f"  * package: {package.get('pypi_name', '')}")
         rendered_url = url.format(repo=package['repo'], pypi_name=package['pypi_name'])
         r = requests.get(rendered_url)
+        content = r.content
+        if badge == 'pypi_downloads' and 'pypi: invalid' in r.content:
+            print("PyPI badge returned as 'invalid'. Retrying after 5 seconds.")
+            time.sleep(5)
+            r = requests.get(rendered_url)
+            content = r.content
+
+        if badge == 'pypi_downloads' and 'pypi: invalid' in r.content:
+            print("PyPI still returning 'invalid' badge after retry.")
```

Comment:
If it's still invalid, it looks like theis script will still return the invalid script. Can you just not return anything, so that it either uses the old one (hopefully not much out of date) or none at all in the cache so that it fetches the original instead at a different level?

Reply:
>  Can you just not return anything, so that it either uses the old one (hopefully not much out of date) 

Yes, that is the easy case to handle. If the old one isn't 'invalid' then it will be used and not overwritten with one that is 'invalid'.

> ... or none at all in the cache so that it fetches the original instead at a different level?

That is a little trickier as you would need at least two steps 1) a way to `git rm` the file from the repo and commit that removal from the cache 2) a fallback so that when the cached version doesn't exist, it uses the live link (this may exist already, but I would need to check).

I'll implement your first suggestion. Would be good to try out this change but the tests are failing for unrelated environment solving reasons that also need to be debugged...